### ISSUE TITLE
the training error will go to Nan by using your default parameter

### ISSUE THREAD

**Issue by gaopeng-eugene** [amdegroot/ssd.pytorch#7]
Loading base network...
Initializing weights...
Loading Dataset...
Training SSD on VOC2007
Timer: 6.7833 sec.
iter 0 || Loss: 26.1034 || Timer: 0.2098 sec.
iter 10 || Loss: 15.1629 || Timer: 0.2115 sec.
iter 20 || Loss: 15.4713 || Timer: 0.2101 sec.
iter 30 || Loss: 17.6274 || Timer: 0.2153 sec.
iter 40 || Loss: 31.7296 || Timer: 0.2107 sec.
iter 50 || Loss: nan || Timer: 0.2113 sec.
iter 60 || Loss: nan || Timer: 0.2073 sec.
iter 70 || Loss: nan || Timer: 0.2035 sec.
iter 80 || Loss: nan || Timer: 0.2090 sec.
iter 90 || Loss: nan || Timer: 0.2055 sec.
iter 100 || Loss: nan || Timer: 0.2196 sec.
iter 110 || Loss: nan || Timer: 0.2064 sec.
iter 120 || Loss: nan || Timer: 0.2257 sec.
iter 130 || Loss: nan || Timer: 0.2051 sec.
iter 140 || Loss: nan || Timer: 0.2142 sec.
iter 150 || Loss: nan || Timer: 0.2056 sec.
iter 160 || Loss: nan || Timer: 0.2122 sec.
iter 170 || Loss: nan || Timer: 0.2090 sec.
iter 180 || Loss: nan || Timer: 0.2091 sec.
iter 190 || Loss: nan || Timer: 0.2110 sec.

**Comment by amdegroot** [comment]
Hi, sorry about that.  It looks like you are using the default values from an older version of the repository.  I accidentally pushed a combination of defaults at one point a few weeks ago, but those have since been updated if you pull the latest version of master.  I need to update the README to reflect that change.  

The issue is that the gradients are much more volatile with smaller batch sizes when using the same learning rate and momentum as used with a batch size of 32.  I think that if you start with a lower learning rate and run for 20k iterations or so before lowering to the default learning rate, you should still be able to see convergence.  If you would like more details on how to train with a batch size of 16, I suggest reading some of the [issues](https://github.com/weiliu89/caffe/issues) posted on Wei Liu's caffe SSD about training with smaller batch sizes.
